### ISSUE TITLE
`T(Inf) == NaN`?

### ISSUE THREAD

**Issue by anriseth** [JuliaMath/DoubleFloats.jl#18]
With `T = DoubleFloat{Float64}`, I get `T(Inf) == NaN`. Is this expected behaviour, or would you want that to become `T(Inf) == convert(T, Inf)` or `T(Inf) == inf(T)`?

This causes problems for example with the `Parameters.jl` package,
```jl
using Parameters, DoubleFloats
@with_kw struct TestMe{T}
   a::T = Inf
end

T = DoubleFloat{Float64}
t = TestMe{T}()
t.a == inf(T) # false
isnan(t.a) # true
```

We have some [code in LineSearches.jl](https://github.com/JuliaNLSolvers/LineSearches.jl/blob/master/src/initialguess.jl#L35) that uses the `Parameters.jl` package and this issue comes up there.

**Comment by JeffreySarnoff** [comment]
thanks for the info -- I will fix this

**Comment by JeffreySarnoff** [comment]
Fixed in v0.1.9 -- merge pending

**Comment by anriseth** [comment]
Thank you :))
